apps/avatax/src/modules/avatax/calculate-taxes/avatax-calculate-taxes-payload-lines-transformer.ts

```diff
@@ -5,6 +5,71 @@ import { AvataxTaxCodeMatches } from "../tax-code/avatax-tax-code-match-reposito
 import { AvataxCalculateTaxesTaxCodeMatcher } from "./avatax-calculate-taxes-tax-code-matcher";
 import { avataxShippingLine } from "./avatax-shipping-line";
 
+type TaxBaseLine = TaxBaseFragment["lines"][0];
+
+const getUndiscountedTotalPrice = (line: TaxBaseLine) => {
+  if (line.sourceLine.__typename === "CheckoutLine") {
+    return line.sourceLine.undiscountedTotalPrice.amount;
+  }
+
+  return line.sourceLine.undiscountedTotalPrice.net.amount;
+};
+
+const checkIfIsDiscountedLine = (isDiscounted: boolean, line: TaxBaseLine) => {
+  if (isDiscounted) {
+    const undiscountedTotalPrice = getUndiscountedTotalPrice(line);
+    const totalPrice = line.totalPrice.amount;
+
+    return totalPrice !== undiscountedTotalPrice;
+  }
+
+  return false;
+};
+
+/**
+ * If entire checkout discount then lines in webhook subscription are not discounted
+ * If one item is discounted then the line has discounted totalPrice
+ */
+const checkDiscounts = (taxBase: TaxBaseFragment, isDiscounted: boolean) => {
+  let hasEntireCheckoutDiscount = false;
+  let hasOncePerOrderVoucher = false;
+  let discountedLinesCount = 0;
+
+  for (const line of taxBase.lines) {
+    const isLineDiscounted = checkIfIsDiscountedLine(isDiscounted, line);
+
+    if (isLineDiscounted) {
+      discountedLinesCount++;
+    }
+  }
+
+  // If none of the lines are discounted, but isDiscounted is true then it's entire checkout discount
+  if (discountedLinesCount === 0 && isDiscounted) {
+    hasEntireCheckoutDiscount = true;
+  }
+
+  if (discountedLinesCount === 1) {
+    hasOncePerOrderVoucher = true;
+  }
+
+  return { hasEntireCheckoutDiscount, hasOncePerOrderVoucher };
+};
+
+const isProductLineDiscounted = (
+  line: TaxBaseLine,
+  isDiscounted: boolean,
+  hasOncePerOrderVoucher: boolean,
+  hasEntireCheckoutDiscount: boolean,
+) => {
```

Comment:
please avoid function with multiple arguments, especially when arguments are primitives

its very easy to confuse order of 3 booleans in the row

instead use object destructuring